### DISCUSSION TITLE
Using a workspace package "singleton" outside of the workspace

### DISCUSSION THREAD

**Discussion by OliverDudgeon** [pnpm/pnpm#4121] [Q&A] [answered]
pnpm allows the option `shared-workspace-lockfile=false` creating lockfiles inside each package. A package might be a git submodule which lives in a different repo but should be integrable in the workspace. This can be motivated by the idea of wanting a public package that lives inside a private "monorepo". If this package depends on other parts of the workspace, the workspace protocol makes the workflow very nice! However, the submodule has a flaw where the lockfile can't be used to `pnpm install` as it expects to find the workspace packages, 

` ERR_PNPM_NO_MATCHING_VERSION_INSIDE_WORKSPACE  In : No matching version found for @squonk/eslint-config@^0.2.0 inside the workspace`

but when the submodule is on its own I'm wanting the packages to be installed from npm. Is there a setup that would get the two environments to install without modifying the `package.json` for each environment?

Instead if pnpm in trying to install the packages and can't find a `pnpm-workspace.yaml` perhaps pnpm should fallback to installing from npm (maybe under a config option)?

**Comment by zkochan** [comment, marked as answer]
You can omit the `workspace:` protocol. The package will be linked from the workspace anyway https://pnpm.io/workspaces#link-workspace-packages

**Reply by OliverDudgeon** [reply]
Thanks that's simple and does what I want.
However, I do seem to be having problems with the install in a CI environment. I'm running `pnpm i` inside a Github actions and the once linked local packages are skipped `WARN  Local dependency not found at /home/runner/work/libs/theme` leading to errors.

**Reply by zkochan** [reply]
try to remove the lockfile

**Reply by OliverDudgeon** [reply]
Weirdly it works fine in a Docker environment but not Github actions
